### DISCUSSION TITLE
json.dumps with sort_keys?

### DISCUSSION THREAD

**Discussion by kakopappa** [micropython/micropython#13235] [Core Development] [answered]
I am porting SinricPro library for Micropython.

```
message = """{"header":{"payloadVersion":2,"signatureVersion":1},"payload":{"action":"setPowerState","clientId":"portal","createdAt":1703039051,"deviceId":"6582437f7818e79186337f1a","replyToken":"c8852568-f871-41ec-b4b7-1bdb83cbf471","type":"request","value":{"state":"On"}},"signature":{"HMAC":"q5Uvkfo5TlMHUvxRtfOLzHrUM0VKEjb1d0Uy240/2y8="}}"""
message_dict = json.loads(message)
print(json.dumps(message_dict["payload"], separators=(',', ':')).encode("utf-8"))
```

prints

`b'{"deviceId":"6582437f7818e79186337f1a","clientId":"portal","createdAt":1703039051,"action":"setPowerState","replyToken":"c8852568-f871-41ec-b4b7-1bdb83cbf471","type":"request","value":{"state":"On"}}'`

`json.dumps` seems to be messing up the keys order  in `payload` and because of that HMAC signature calculation generates the wrong signature. I have seen an option called `sort_keys`  but it seems not available in Micropython. 

Any workaround for this ?

**Comment by andrewleech** [comment, marked as answer]
It's a bit messy but you can turn the dict into a sorted OrderedDict first, I think this would work:
```
from collections import OrderedDict


message_dict = json.loads(message) 
message_dict_s = OrderedDict(sorted(message_dict.items()))
```
